lib/charms/blackbox_k8s/v0/blackbox_probes.py

```diff
@@ -0,0 +1,652 @@
+An example of defining a module is:
+
+modules={
+            "http_2xx_longer_timeout": {
+                "prober": "http"
+                "timeout": "30s"  # default is 5s
+            }
+        }
+
+## Consumer Library Usage
+
+The `BlackboxProbesRequirer` object may be used by the Blackbox Exporter
+charms to retrieve the probes to be monitored. For this
+purposes a Blackbox Exporter charm needs to do two things:
+
+1. Instantiate the `BlackboxProbesRequirer` object by providing it a
+reference to the parent (Blackbox Exporter) charm and, optionally, the name of
+the relation that the Blackbox Exporter charm uses to interact with probes
+targets. This relation must confirm to the `blackbox_probes`
+interface and it is strongly recommended that this relation be named
+`blackbox-probes` which is its default value.
+
+For example a Blackbox Exporter may instantiate the
+`BlackboxProbesRequirer` in its constructor as follows
+
+    from charms.blackbox_k8s.v0.blackbox_probes import BlackboxProbesRequirer
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        ...
+        self.probes_consumer = BlackboxProbesRequirer(
+            charm=self,
+            relation_name="blackbox-probes",
+        )
+        ...
+
+The probes consumer must be instantiated before the prometheus_scrape MetricsEndpoint Provider,
+because Blackbox defines new metrics endpoints to send to Prometheus.
+
+2. A Blackbox Exporter charm also needs to respond to the
+`TargetsChangedEvent` event of the `BlackboxProbesRequirer` by adding itself as
+an observer for these events, as in
+
+    self.framework.observe(
+        self.probes_consumer.on.targets_changed,
+        self._on_scrape_targets_changed,
+    )
+
+In responding to the `TargetsChangedEvent` event the Blackbox Exporter
+charm must update its configuration so that any new probe
+is added and/or old ones removed from the list.
+For this purpose the `BlackboxProbesRequirer` object
+exposes a `probes()` method that returns a list of probes jobs. Each
+element of this list is a probes configuration to be added to the list of jobs for
+Prometheus to monitor. 
+Same goes for the list of client charm defined modules. The `BlackboxProbesRequirer` object
+exposes a `modules()` method that returns a dict of the new modules to be added to the
+Blackbox configuration file.
+"""
+
+import logging
+import json
+from typing import Dict, List, Optional, Union, MutableMapping
+
+from ops import Object
+from ops.charm import CharmBase
+from ops.framework import BoundEvent, EventBase
+from ops.charm import CharmBase, RelationRole
+from cosl import JujuTopology
+from ops.model import ModelError, Relation
+from ops.framework import (
+    BoundEvent,
+    EventBase,
+    EventSource,
+    Object,
+    ObjectEvents,
+)
+import pydantic
+from pydantic import BaseModel, ConfigDict, Field, Json
+
+# The unique Charmhub library identifier, never change it
+LIBID = "1"
```

Comment:
I think you need to run `charmcraft create-lib` in order to get the unique identifier assigned by Charmhub